From 3b87ef68eef108d6bd85511586fc01824ab42475 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:44:12 -0400
Subject: [acpi] mm: use NUMA_NO_NODE

Message-id: <20140806034412.1438.21829.stgit@gir.stowe>
Patchwork-id: 87410
O-Subject: [RHEL7.1 PATCH 01/69] ACPI / mm: use NUMA_NO_NODE
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 1bb25df0fde2cdb2f250a7e7e43c2ec1ba65d0f5

commit 1bb25df0fde2cdb2f250a7e7e43c2ec1ba65d0f5
Author: Jianguo Wu <wujianguo@huawei.com>
Date:   Fri Aug 30 09:25:40 2013 +0800

    ACPI / mm: use NUMA_NO_NODE

    Use more appropriate NUMA_NO_NODE instead of -1

    Signed-off-by: Jianguo Wu <wujianguo@huawei.com>
    Acked-by: David Rientjes <rientjes@google.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/acpi_memhotplug.c b/drivers/acpi/acpi_memhotplug.c
index 6c67cc4..383cd80 100644
--- a/drivers/acpi/acpi_memhotplug.c
+++ b/drivers/acpi/acpi_memhotplug.c
@@ -282,7 +282,7 @@ static void acpi_memory_remove_memory(struct acpi_memory_device *mem_device)
   if (!info->enabled)
    continue;
 
-  if (nid < 0)
+  if (nid == NUMA_NO_NODE)
    nid = memory_add_physaddr_to_nid(info->start_addr);
 
   acpi_unbind_memory_blocks(info, handle);
diff --git a/drivers/acpi/numa.c b/drivers/acpi/numa.c
index 33e609f..09f79a2 100644
--- a/drivers/acpi/numa.c
+++ b/drivers/acpi/numa.c
@@ -73,7 +73,7 @@ int acpi_map_pxm_to_node(int pxm)
 {
  int node = pxm_to_node_map[pxm];
 
- if (node < 0) {
+ if (node == NUMA_NO_NODE) {
   if (nodes_weight(nodes_found_map) >= MAX_NUMNODES)
    return NUMA_NO_NODE;
   node = first_unset_node(nodes_found_map);
@@ -334,7 +334,7 @@ int acpi_get_pxm(acpi_handle h)
 
 int acpi_get_node(acpi_handle *handle)
 {
- int pxm, node = -1;
+ int pxm, node = NUMA_NO_NODE;
 
  pxm = acpi_get_pxm(handle);
  if (pxm >= 0 && pxm < MAX_PXM_DOMAINS)
-- 
1.7.1